runtime/help/plugins.md

```diff
@@ -51,6 +50,9 @@ which micro defines:
 
 * `postinit()`: initialization function called after `init()`.
 
+* `deinit()`: cleanup function called when your plugin is unloaded,
+   either manually by the user, or by running the `reload` command.
```

Comment:
It's not clear what does it mean "unloading manually" (I ended up looking into the code to recall what it means).

So, it means "disabling plugin via the `set` command", right?

BTW, the `reload` command is also run by the user, and also manually, right? :)

BTW `preinit()`, `init()` and `postinit()` are also called when running the `reload` command. It doesn't mean that we need to mention it in their above descriptions as well, but we should probably better document the `reload` command itself, in `help commands`. Currently it is:
```
* `reload`: reloads all runtime files.
```

It's likely not clear to the user what is "runtime" files. So maybe let's make it clear:

```
* `reload`: reloads all runtime files (settings, keybindings, syntax files,
   colorschemes, plugins).
```

And then maybe also document its behavior wrt callbacks, something like:

```
* `reload`: reloads all runtime files (settings, keybindings, syntax files,
   colorschemes, plugins). For plugins, it properly unloads already loaded
   plugins (by running their `deinit()` callbacksif there are any) and then
   re-initializes reloaded plugins (by running their `preinit()`, `init()`
   and `postinit()` callbacks if there are any).
```

Please feel free to suggest a better wording.

Reply:
> `help options`
> 
> > [...]
> > Plugin options: all plugins come with a special option to enable or disable
> > them. The option is a boolean with the same name as the plugin itself.
> > [...]

Ah, indeed. So it's actually documented, so it's not so bad. So maybe it's not necessary to lengthen the description of the `set` command with an explicit mentioning of plugin options. Still, I guess we should add mentioning them at least to `help plugins`, since it is, well, about plugins. (I.e. `help plugins` is probably the place where users would look for info how to disable a plugin and so on.)

Reply:
I had some trouble figuring out how to naturally fit this into `plugins.md` -- I ended up just adding a note to the top of the file, clarifying that `plugins.md` describes how to *create* plugins, and then directed the user to either `help options` or `help commands`, depending on what they're looking for. Hopefully that is good enough, unless you have a better idea?

Reply:
LGTM, thanks.